lexxpluss_apps/src/zcan_actuator.hpp

```diff
@@ -25,65 +25,59 @@
 
 #pragma once
 
-#include <zephyr.h>
-#include <device.h>
-#include <drivers/gpio.h>
-#include <drivers/can.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/can.h>
+#include <zephyr/logging/log.h>
 #include "actuator_controller.hpp"
 
-#define CAN_ID_ACTUATOR_CONTROL 0x208 // based on saito-san's CAN ID assignment
-#define CAN_ID_ACTUATOR_ENCODER 0x209 // based on saito-san's CAN ID assignment
-#define CAN_ID_ACTUATOR_CURRENT 0x20a // based on saito-san's CAN ID assignment
+#define CAN_ID_ACTUATOR_CONTROL 0x208 // based on CAN ID assignment
+#define CAN_ID_ACTUATOR_ENCODER 0x209 // based on CAN ID assignment
+#define CAN_ID_ACTUATOR_CURRENT 0x20a // based on CAN ID assignment
 #define CAN_DATALENGTH_ACTUATOR_ENCODER 6
 #define CAN_DATALENGTH_ACTUATOR_CURRENT 8
 
+namespace lexxhard::zcan_actuator {
 
-
-namespace lexxhard {
-
+LOG_MODULE_REGISTER(zcan_actuator);
 char __aligned(4) msgq_can_actuator_control_buffer[8 * sizeof(actuator_controller::can_format_control)];
-k_msgq msgq_can_actuator_control;
 
-class ros_actuator {
+CAN_MSGQ_DEFINE(msgq_can_actuator_control, 16);
+
+class zcan_actuator {
 public:
-    int init() {
+    void init() {
         // can device bind
         k_msgq_init(&msgq_can_actuator_control, msgq_can_actuator_control_buffer, sizeof(actuator_controller::can_format_control), 8);
-        dev = device_get_binding("CAN_2");
-        if (!device_is_ready(dev))
-            return -1;
+        dev = DEVICE_DT_GET(DT_NODELABEL(can2));
+        if (!device_is_ready(dev)){
+            LOG_INF("CAN_2 is not ready");
```

Comment:
For warning messages, there was a change from LOG_INF to LOG_ERR; it may be more consistent to use LOG_ERR for cases where can2 cannot be retrieved.

警告メッセージについて、LOG_INFからLOG_ERRに変更している箇所がありました。can2が取得できない場合にもLOG_ERRとする方が統一感があるかもしれません。

Reply:
デバイス取得できない明らかなエラーは他の部分も含めてLOG_ERRにしました。

Obvious errors that prevent device acquisition are made LOG_ERR, including other parts.

Reply:
https://github.com/LexxPluss/LexxHard-SensorControlBoard-Firmware/commit/63493394828cf0503c491efdabaa6214110ae996